File: frontend/src/utils/keepAlive.js
// Keep backend alive by pinging every 10 minutes
const BACKEND_URL = 'https://rwanda-edu-platform.onrender.com'
const PING_INTERVAL = 10 * 60 * 1000 // 10 minutes

let pingInterval = null

export function startKeepAlive() {
  if (pingInterval) return
  
  const ping = async () => {
    try {
      await fetch(`${BACKEND_URL}/health`, { method: 'GET' })
    } catch (error) {
      // Silent fail
    }
  }
  
  ping()
  pingInterval = setInterval(ping, PING_INTERVAL)
}

export function stopKeepAlive() {
  if (pingInterval) {
    clearInterval(pingInterval)
    pingInterval = null
  }
}

File: backend/keep_alive.py
import requests
import time
import os
from datetime import datetime

BACKEND_URL = os.getenv('BACKEND_URL', 'https://your-backend-url.com')
PING_INTERVAL = 600  # 10 minutes in seconds

def ping_backend():
    try:
        response = requests.get(f"{BACKEND_URL}/api/health", timeout=10)
        status = "✓ UP" if response.status_code == 200 else f"✗ DOWN ({response.status_code})"
        print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] Backend {status}")
        return True
    except Exception as e:
        print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] Backend ✗ ERROR: {str(e)}")
        return False

if __name__ == "__main__":
    print(f"Keep-alive service started. Pinging {BACKEND_URL} every {PING_INTERVAL//60} minutes...")
    while True:
        ping_backend()
        time.sleep(PING_INTERVAL)
